flytekit/contrib/task.py

```diff
@@ -0,0 +1,60 @@
+from datetime import timedelta as _timedelta
+
+from flytekit.common import constants as _common_constants
+from flytekit.common.exceptions import user as _user_exceptions
+from flytekit.common.tasks import sdk_runnable as _sdk_runnable
+
+from flytekit.contrib.base_sensor import Sensor as _Sensor
+
+
+class SensorTask(_sdk_runnable.SdkRunnableTask):
+
+    def __init__(self, **kwargs):
+        super(SensorTask, self).__init__(**kwargs)
+
+    def _execute_user_code(self, context, inputs):
+        sensor = super()._execute_user_code(context=context, inputs=inputs)
+        if sensor is not None:
+            if not isinstance(sensor, _Sensor):
+                raise _user_exceptions.FlyteTypeException(
+                    received_type=type(sensor),
+                    expected_type=type(_Sensor),
+                )
+            succeeded = sensor.sense()
+            if not succeeded:
+                raise _user_exceptions.FlyteRecoverableException()
+
+
+def sensor_task(
+    _task_function=None,
```

Comment:
nit: For consistency, let's not set non-empty/non-zero defaults in code.  And since this really is just a special case of a python task, we should support all the same kwargs (as shown here: https://github.com/lyft/flytekit/blob/master/flytekit/sdk/tasks.py#L114-L129)

Also, let's give a descriptive docstring describing how it works.